kernel/CMakeLists.txt

```diff
@@ -58,6 +58,7 @@ target_sources_ifdef(
   mem_domain.c
   userspace_handler.c
   userspace.c
+  cache_handlers.c
```

Comment:
I do not think this belongs under kernel/, this is not part of the kernel and should probably be part of the arch/?

Reply:
not following you here. to me arch/ is the architecture-specific parts of kernel/. if it's not arch-specific it doesn't go there. And we are exposing this functionality as a system call, which is a kernel entry point.

Reply:
want to see the handlers where the code is, in this case it is in arch and declared in its own header, so it is not a "kernel" (in my view, is what is really implemented under kernel/) and is rather an os/architecture feature.
Any reason why this cant go into arch/common/?

Besides, based on the comment 4 lines above this change, the ordering is wrong if we are to keep it in this location.

Reply:
> "kernel" (in my view, is what is really implemented under kernel/) and is rather an os/architecture feature.

I'm not fully understanding the difference in semantics from "os" and "kernel" here. This is implementing interfaces defined at the kernel level. If these weren't inline functions they would need the same C file.

I also don't understand why "arch/common" exists other than past bikeshedding. arch/ is supposed to be for the parts of the kernel which require different implementations based on CPU type

I really do not want to go down a path where we have some arbitrary distinction between "kernel/" and "arch/common/" that nobody can precisely explain. IMO everything in arch/common belongs in the kernel

Reply:
just like we have drivers/i2c/i2c_handlers.c where the i2c syscalls are implemented and not under kernel/, for the sake of having things organized, put the handlers at the same level where the implementation is.

Reply:
but the implementation is in `include/sys/`. I don't see what `arch/common/` has to do with this. The sycalls for include/sys/atomic.c are implemented in the kernel. Every syscall in `include/sys` is implemented in either `kernel` or `lib/os` as far as I can see.

You might argue, "put it in lib/os" but we shouldn't be putting syscalls there either, the ones that are in there need to be moved to the kernel. When we eventually support separate kernel/application, the lib/os stuff will all go in the application and not the kernel's memory space

Reply:
> Besides, based on the comment 4 lines above this change, the ordering is wrong if we are to keep it in this location.

Addressed this. The ordering of cache_handlers.h is before the last two files now in kernel/CMakeLists.txt.